spring-xd-analytics/src/main/java/org/springframework/xd/analytics/metrics/redis/RedisRichGaugeRepository.java

```diff
@@ -26,14 +35,17 @@
  * Repository for rich-gauges backed by Redis.
  *
  * @author Luke Taylor
+ * @author Eric Bottard
  */
-public final class RedisRichGaugeRepository extends
+public class RedisRichGaugeRepository extends
 		AbstractRedisMetricRepository<RichGauge, String> implements RichGaugeRepository {
 
-	private static final String ZERO = serialize(new RichGauge("zero"));
+	private static final String ZERO = serialize(new RichGauge("ZERO"));
+
+	private RetryTemplate retryTemplate = new RetryTemplate();
```

Comment:
I think we should expose the options for `ExponentialBackOffPolicy` and max number of retries instead as in 

https://github.com/spring-projects/spring-xd/blob/master/spring-xd-messagebus-spi/src/main/java/org/springframework/xd/dirt/integration/bus/MessageBusSupport.java#L751

The defaults in `ExponentialBackOffPolicy` may need to be tweeked.  It probably doesn't make sense to try and write if there is contention for the key more than 5 seconds or so.  The current retry as fast as possible for 3x is probably too aggressive.